core/src/test/scala/org/apache/spark/sql/store/TokenizationTest.scala

```diff
@@ -488,6 +489,28 @@ class TokenizationTest
     }
   }
 
+  test("Test BUG SNAP-1642") {
+    SnappyTableStatsProviderService.suspendCacheInvalidation = true
+    val maxquery = s"select * from $table where a = (select max(a) from $table)"
+    val numRows = 10
+    createSimpleTableAndPoupulateData(numRows, s"$table", true)
+
+    val rs1 = snc.sql(maxquery)
+    val rows1 = rs1.collect()
```

Comment:
Main idea is to check whether cached and uncached are same or not. So validating that below.